pkg/tools/fs/utils_test.go

```diff
@@ -26,55 +27,89 @@ import (
 	"testing"
 	"time"
 
+	"github.com/google/uuid"
 	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
 
 	"github.com/networkservicemesh/sdk/pkg/tools/fs"
 )
 
+const macOSName = "darwin"
+
+func checkPathError(t *testing.T, err error) error {
+	if fErr, ok := err.(*iofs.PathError); ok {
+		if u := fErr.Unwrap(); u != nil {
+			t.Log(u) // Without unwrapping printing gives only numeric error code.
+		}
+	}
+
+	return err
+}
+
 func Test_WatchFile(t *testing.T) {
-	t.Skip("https://github.com/networkservicemesh/sdk/issues/848")
+	t.Cleanup(func() { goleak.VerifyNone(t) })
 
 	root := filepath.Join(os.TempDir(), t.Name())
+
+	path := filepath.Join(root, uuid.New().String())
+	err := os.MkdirAll(path, os.ModePerm)
 	defer func() {
-		_ = os.RemoveAll(root)
+		_ = os.RemoveAll(path)
 	}()
-
-	path := filepath.Join(root, "A")
-	_ = os.MkdirAll(path, os.ModePerm)
+	require.NoError(t, checkPathError(t, err))
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
 	filePath := filepath.Join(path, "file1.txt")
 	ch := fs.WatchFile(ctx, filePath)
 
-	expectEvent := func(assertFunc func(require.TestingT, interface{}, ...interface{})) {
+	expectEvent := func() []byte {
 		select {
 		case <-time.After(time.Second):
```

Comment:
Actually, this is not correct to use a timeout in `1s` for each event. Please let's set one timeout in `1s` for all events.
```suggestion
		case <-timeoutCh:
```